networkx/algorithms/approximation/broadcasting.py

```diff
@@ -0,0 +1,98 @@
+import networkx as nx
+from networkx.utils import not_implemented_for
+
+__all__ = [
+    "tree_broadcast",
+]
+
+
+def update_broadcast_label(G, U, v):
+    adj = [u for u in G.neighbors(v) if u in U]
+    adj.sort(key=lambda u: G.nodes[u]["value"], reverse=True)
+    return max([G.nodes[u]["value"] + i for i, u in enumerate(adj, start=1)])
+
+
+@not_implemented_for("directed")
+@not_implemented_for("multigraph")
+def tree_broadcast(G):
```

Comment:
Since the value returns in a time, this should probably be named `tree_broadcast_time()` (or similar)

Reply:
done